test/cocotb/osdtestlib/eval_utility.py

```diff
@@ -4,59 +4,87 @@
 from cocotb.result import TestFailure
 from cocotb.triggers import RisingEdge
 
+types = ['REG', 'PLAIN', 'EVENT']
+
+subtypes = ['REQ_READ_REG_16', 'REQ_READ_REG_32', 'REQ_READ_REG_64',
+            'REQ_READ_REG_128', 'RESP_READ_REG_SUCCESS_16',
+            'RESP_READ_REG_SUCCESS_32', 'RESP_READ_REG_SUCCESS_64',
+            'RESP_READ_REG_SUCCESS_128', 'RESP_READ_REG_ERROR',
+            'REQ_WRITE_REG_16', 'REQ_WRITE_REG_32', 'REQ_WRITE_REG_64',
+            'REQ_WRITE_REG_128', 'RESP_WRITE_REG_SUCCESS',
+            'RESP_WRITE_REG_ERROR']
```

Comment:
You added these two lists to get a mapping from the constant to the names for the types/subtypes? This is rather prone to bugs in the future, if the mapping should changes. A much nicer way is to make the constants an Enum already in DiPacket or a related class (see here: https://docs.python.org/3/library/enum.html). This gives you the functionality you're trying to create here for free.